Cython/Compiler/Code.py

```diff
@@ -1557,6 +1559,21 @@ def generate_object_constant_decls(self):
             self.parts['module_state_traverse'].putln(
                 "Py_VISIT(traverse_module_state->%s);" % cname)
 
+        for prefix, count in self.array_consts.items():
+            full_prefix = f"{Naming.pyrex_prefix}{prefix}"
+            self.parts['module_state'].putln(f"PyObject *{full_prefix}[{count}];")
+            self.parts['module_state_defines'].putln(
+                f"#define {full_prefix} {Naming.modulestateglobal_cname}->{full_prefix}")
+            cleanup_details = [('module_state_clear', 'Py_CLEAR', 'clear_module_state->'),
+                               ('module_state_traverse', 'Py_VISIT', 'traverse_module_state->')]
+            if Options.generate_cleanup_code >= 2 and prefix != 'ustring':
+                cleanup_details.append(('cleanup_globals', 'Py_CLEAR', ''))
```

Comment:
I think this would be cleaner with a global dict that maps prefixes to cleanup code levels. Even if it's just "usting=None, everything else =2", then it would at least be visible in one place.